WE-468 include requested rig attributes

LouisLinnerud opened the pull request:
## Fixes
This pull request fixes #169

## Description
Extended already existing support in: 
- witsml.api/models/Rig.cs 
- witsml/data/witsmlRig.cs 
- Witsml.api/query/RigQueries.cs 
- witsml.api/services/rigservice.cs
to accommodate more attributes
...

## Type of change
_List the types of change. Remove from the list any points which are not necessary_

* Enhancement of existing functionality


## Impacted Areas in Application
- witsml.api/models/Rig.cs 
- witsml/data/witsmlRig.cs 
- Witsml.api/query/RigQueries.cs 
- witsml.api/services/rigservice.cs

## Checklist:
_Please tick all the boxes or remove the ones that aren't needed and explain why_

*Communication*
* [x] PR is related to an issue


*Code quality*
* [x] Code follows the style guidelines
* [x] I have self-reviewed my code
* [x] No new warnings are generated

*Test coverage*
* [x] Existing tests pass


## Further comments

LouisLinnerud committed:
Merge pull request #1 from LouisLinnerud/#1297-remove-EC-lint-reference

replace EClint reference with dotnet

Commit:
Merge branch 'main' of github.com:equinor/witsml-explorer

Commit:
Merge branch 'we-464' of https://github.com/eirinsvi/witsml-explorer

Commit:
Merge branch 'main' of github.com:equinor/witsml-explorer

Commit:
Merge remote-tracking branch 'upstream/main'

Commit:
Merge branch 'main' of github.com:equinor/witsml-explorer

LouisLinnerud committed:
Extend back end suppert for Rig object

LouisLinnerud committed:
Extend queries and support in frontend

janburak commented on Src/Witsml/Data/WitsmlRig.cs:
```diff
@@ -25,6 +28,59 @@ public class WitsmlRig
         [XmlElement("typeRig")]
         public string TypeRig { get; set; }
 
+        [XmlElement("DTimCreation")]
```
DTimCreation and DTimLastChange are part of CommonData and should not be included directly here

LouisLinnerud replied:
👍

janburak commented on Src/Witsml/Data/WitsmlRig.cs:
```diff
@@ -25,6 +28,59 @@ public class WitsmlRig
+        }
+
+        [XmlElement("itemState")]
+        public string ItemState { get; set; }
```
ItemState is part of CommonData and should not be included directly here

LouisLinnerud replied:
👍

janburak commented on Src/WitsmlExplorer.Frontend/components/ContentViews/RigsListView.tsx:
```diff
@@ -13,15 +13,44 @@ export const RigsListView = (): React.ReactElement => {
+    { property: "airGap", label: "AirGap", type: ContentType.String },
+    { property: "dTimStartOp", label: "DateTimeStartOperating", type: ContentType.DateTime },
+    { property: "dTimEndOp", label: "DateTimeEndOperating", type: ContentType.DateTime },
+    { property: "dTimCreation", label: "DTimCreation", type: ContentType.String },
```
I think dTimCreation and dTimLastChange can be removed form the content view given that they are not mentioned in the issue

LouisLinnerud replied:
👍

janburak commented on Src/WitsmlExplorer.Frontend/models/rig.tsx:
```diff
@@ -1,31 +1,57 @@
+
 export default interface Rig {
-  airGap: string;
+  airGap: Measure;
```
Inconsistent use of letter casing. Use lowercase.

LouisLinnerud replied:
I believe it has to be as it is, else it will be troublesome to load the xml document properly

LouisLinnerud committed:
Fix to feedback on prev PR

LouisLinnerud committed:
refactoring rig interface var names to lower case

LouisLinnerud committed:
now it is lowercase....

LouisLinnerud committed:
lowercase for yearEntService

LouisLinnerud committed:
remove  isOffshoreText from where its unnecessary

janburak commented on Src/WitsmlExplorer.Api/Models/Rig.cs:
```diff
@@ -1,22 +1,35 @@
+        public string EmailAddress { get; set; }
+        public string FaxNumber { get; set; }
+        public bool IsOffshore { get; set; }
+        public string IsOffshoreText { get; set; }
```
IsOffshoreText is unnecessary here

LouisLinnerud replied:
👍

janburak commented on Src/WitsmlExplorer.Api/Models/Rig.cs:
```diff
@@ -1,22 +1,35 @@
+        public DateTime? DTimEndOp { get; set; }
+        public string EmailAddress { get; set; }
+        public string FaxNumber { get; set; }
+        public bool IsOffshore { get; set; }
```
This should be "bool? IsOffshore" because it can be null

LouisLinnerud replied:
👍

janburak commented on Src/WitsmlExplorer.Api/Services/RigService.cs:
```diff
@@ -47,18 +64,30 @@ public async Task<Rig> GetRig(string wellUid, string wellboreUid, string rigUid)
+                DTimEndOp = StringHelpers.ToDateTime(witsmlRig.DTimEndOp),
+                EmailAddress = witsmlRig.EmailAddress,
+                FaxNumber = witsmlRig.FaxNumber,
+                IsOffshore = bool.Parse(witsmlRig.IsOffshoreText),
```
sufficient to set equal to "witsmlRig.IsOffshore"

LouisLinnerud replied:
isOffshore is a bool so isOffshoreText  has to be parsed, what do you think?

janburak commented on Src/WitsmlExplorer.Frontend/components/ContentViews/RigsListView.tsx:
```diff
@@ -13,15 +13,43 @@ export const RigsListView = (): React.ReactElement => {
+        ratingDrillDepth: `${rig.ratingDrillDepth?.value ?? ""} ${rig.ratingDrillDepth?.uom ?? ""}`,
+        ratingWaterDepth: `${rig.ratingWaterDepth?.value ?? ""} ${rig.ratingWaterDepth?.uom ?? ""}`,
+        airGap: `${rig.airGap?.value ?? ""} ${rig.airGap?.uom ?? ""}`,
+        isOffshore: `${rig.isOffshore.toString()}`,
```
isOffshore is defined as string so it is redundant to call toString()

LouisLinnerud replied:
isOffshore is a bool so it has to be parsed, what do you think?

LouisLinnerud committed:
Fix bug for null value in "isOffshore"
resolved to the old solution where "IsOffshoreText" is used to pass the true value to "IsOffshore"
also changing "IsOffshore" from bool to string in Rig.cs

LouisLinnerud committed:
remove unnecessary variable declarations in Rig.cs

LouisLinnerud committed:
Fix correct type for attribute "isOffshore"
